map-storage/src/Upload/FileSystemInterface.ts

```diff
@@ -11,7 +11,11 @@ export interface FileSystemInterface {
 
     readFileAsString(virtualPath: string): Promise<string>;
 
+    listFiles(virtualDirectory: string, extension?: string): Promise<string[]>;
+
     writeStringAsFile(virtualPath: string, content: string): Promise<void>;
 
+    writeCache(virtualPath: string, fileName: string, content: string): Promise<void>;
```

Comment:
I don't think this method is necessary in fact.

It does not write a "cache", it simply writes a file. And we already have a method for that: `writeStringAsFile`.
And if you look closely at `writeStringAsFile` and `writeCache`, you will find out they are almost identical.

I think you should remove `writeCache` method completely and potentially adapt the `writeStringAsFile` if needed (I think `writeCache` is creating the directory if it does not exists and `writeStringAsFile` is not).

I think you should

Reply:
I separated it as the previous version was saving cache file on the top level and paths were different. Now this is not the case and it can in fact be unified :) Thanks for the comment!